packages/ember-glimmer/tests/integration/syntax/if-unless-test.js

```diff
@@ -24,3 +28,85 @@ moduleFor('Syntax test: {{#if}} and {{#unless}} without inverse', class extends
   }
 
 });
+
+moduleFor('Syntax test: {{#if}}', class extends RenderingTest {
+
+  ['@test child conditional should not render children if parent conditional becomes false'](assert) {
```

Comment:
Nice idea, I've moved it to `TogglingSyntaxConditionalsTest`. I didn't manage to get it to work in `TogglingConditionalsTest`, I've added this to my list of things to revisit soon though